File: php/features.php
<?php
declare(strict_types=1);

// Disney SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class DisneyFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new DisneyBaseFeature();
            case "test":
                return new DisneyTestFeature();
            default:
                return new DisneyBaseFeature();
        }
    }
}

File: py/disney_sdk/features.py
# Disney SDK feature factory

from disney_sdk.feature.base_feature import DisneyBaseFeature
from disney_sdk.feature.test_feature import DisneyTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DisneyBaseFeature(),
        "test": lambda: DisneyTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
